src/main/resources/sql/CreateAppDb.sql

```diff
@@ -95,6 +95,8 @@ CREATE TABLE IF NOT EXISTS resource_link (
     CONSTRAINT prevent_both_not_null
         CHECK ((collection_fk is NULL) or (content_fk is NULL))
 );
+CREATE INDEX IF NOT EXISTS idx_resource_link_collection ON resource_link (collection_fk);
+CREATE INDEX IF NOT EXISTS idx_resource_link_content ON resource_link (content_fk);
```

Comment:
I'm always embarrassingly slow to realize a tiny DB tweak is worth many hours of code optimizations.  These lines dropped import time (with the new subtreeHasResources lookup) from many minutes back down to seconds.